### PR TITLE
Added ProfileTab utility class

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

none

#### What's this PR do?

this creates a utility class called `ProfileTab` which lets us cut down on boilerplate a little (when setting up the profile tabs).

we just write components that inherit from `ProfileTab`, and the `boundTextField` and so on functions are already bound to `this` (just remember to call `super(props)`in your constructor!).

#### Where should the reviewer start?

read the new class (`static/js/util/ProfileTab.js) and review the changes to the personal tab (`static/js/components/PersonalTab.js`).

#### How should this be manually tested?

Make sure this doesn't clobber any existing behavior.

#### Any background context you want to provide?

my name is alice and I hate boilerplate.

### CODE REVIEW

File: static/js/components/PersonalTab.js [hunk 1]
```diff
@@ -25,26 +17,22 @@ class PersonalTab extends React.Component {
...
-        )}<br />
+      {this.boundTextField("last_name", "Family name", errors.last_name)}<br />
+      {this.boundTextField("preferred_name", "Preferred name", errors.preferred_name)}<br />
+      {this.boundSelectField('gender', 'Gender', this.genderOptions, errors.gender)}<br />
+      {this.boundSelectField(
+        'preferred_language',
+        'Preferred language',
+        this.languageOptions,
+        errors.preferred_language
+      )}<br />
```
Comment (aliceriot) [inline static/js/components/PersonalTab.js]: this is unrelated, I just noticed that the indentation wasn't correct.

File: static/js/util/ProfileTab.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class ProfileTab extends React.Component {
+  constructor(props) {
+    super(props);
+    this.boundTextField = boundTextField.bind(this);
+    this.boundSelectField = boundSelectField.bind(this);
+    this.boundDateField = boundDateField.bind(this);
+    this.countryOptions = COUNTRIES.map(country => ({
+      value: country.Code,
+      label: country.Name
```
Comment (aliceriot) [inline static/js/util/ProfileTab.js]: I pulled `countryOptions` in here since I think we'll also want these options on both the employment and education tabs.

File: static/js/components/PersonalTab.js [hunk 3]
```diff
@@ -25,26 +17,22 @@ class PersonalTab extends React.Component {
       value: language.alpha2,
       label: language.English
     }));
-    this.countryOptions = COUNTRIES.map(country => ({
-      value: country.Code,
-      label: country.Name
-    }));
     this.saveAndContinue = saveAndContinue.bind(this, '/dashboard');
```
Comment (aliceriot) [inline static/js/components/PersonalTab.js]: this we bind with a specific argument, so we still need to bind it in each tab's constructor.

we could instead make `saveAndContinue` push a `nextTabUrl` prop onto the history, and then it could be bound in `ProfileTab` instead of here, but eh, same amount of code either way (still would have to declare what the `nextTabUrl` prop looked like on every tab component).

**Comment by noisecapella** [comment]
Looks nice and simple :+1:
